pkg/aggregator/statsd.go

```diff
@@ -0,0 +1,207 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"time"
+
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/metrics/event"
+	"github.com/DataDog/datadog-agent/pkg/metrics/servicecheck"
+	taggertypes "github.com/DataDog/datadog-agent/pkg/tagger/types"
+)
+
+// NewStatsdDirect creates a direct interface to the dogstatsd demultiplexer, but exposing the statsd.ClientInterface
+func NewStatsdDirect(demux DemultiplexerWithAggregator) (ddgostatsd.ClientInterface, error) {
+	eventsChan, serviceCheckChan := demux.GetEventsAndServiceChecksChannels()
+	return &statsdDirect{
+		demux:            demux,
+		eventsChan:       eventsChan,
+		serviceCheckChan: serviceCheckChan,
+	}, nil
+}
+
+var _ ddgostatsd.ClientInterface = (*statsdDirect)(nil)
+
+type statsdDirect struct {
+	demux            DemultiplexerWithAggregator
+	eventsChan       chan []*event.Event
+	serviceCheckChan chan []*servicecheck.ServiceCheck
+}
+
+func (s statsdDirect) Gauge(name string, value float64, tags []string, rate float64) error {
+	s.demux.AggregateSample(metrics.MetricSample{
+		Name:       name,
+		Value:      value,
+		Mtype:      metrics.GaugeType,
+		Tags:       tags,
+		SampleRate: rate,
+		Timestamp:  float64(time.Now().Unix()),
+	})
+	return nil
+}
```

Comment:
The main problem with this is that we're not attaching origin information, which means no origin enrichment: no more `kube_pod_name` tag, etc.

I'll have to think through this a little bit and/or chat with some folks about it. Might be an easy way to generate the right origin information and always attach it, but I just don't have the suggestion/solution at my fingertips right now.